src/core/scene/a-scene.js

```diff
@@ -118,7 +118,12 @@ module.exports.AScene = registerElement('a-scene', {
         window.addEventListener('vrdisplaypresentchange', this.onVRPresentChangeBound);
 
         // Enter VR on `vrdisplayactivate` (e.g. putting on Rift headset).
-        window.addEventListener('vrdisplayactivate', function () { self.enterVR(); });
+        window.addEventListener('vrdisplayactivate', function () {
```

Comment:
What about having enterVR check if there is no `this.effect`, and registering an event listener to call itself when its ready.